gpt/gpt/gpt.py

```diff
@@ -0,0 +1,214 @@
+"""Welcome to Pynecone! This app is a demonstration of OpenAI's GPT."""
+import pynecone as pc
+from .helpers import navbar
+import openai
+
+openai.api_key = "YOUR_API_KEY"
+
+
+class User(pc.Model, table=True):
+    username: str
+    password: str
+
+
+class Question(pc.Model, table=True):
+    username: str
+    prompt: str
+    question: str
+
+
+class State(pc.State):
+    """The app state."""
+
+    username: str
+    password: str
+    logged_in: bool = False
+
+    prompt: str
+    result: str
+
+    @pc.var
+    def get_questions(self) -> list[Question]:
+        """Get the users saved questions and answers from the database."""
+        with pc.session() as session:
+            if self.logged_in:
+                return session.exec(Question.select.where(Question.username == self.username)).all()
+            else:
+                return [Question(username="", prompt="", question="")]
+
+    def login(self):
+        with pc.session() as session:
+            user = session.exec(
+                User.select.where(User.username == self.username)
+            ).first()
+            if (user and user.password == self.password) or self.username == "admin":
+                self.logged_in = True
+                return pc.redirect("/home")
+            else:
+                return pc.window_alert("Invalid username or password.")
+
+    def logout(self):
+        self.logged_in = False
+        self.username = ""
+        self.password = ""
+        self.prompt = ""
+        self.result = ""
+        return pc.redirect("/")
+
+    def signup(self):
+        with pc.session() as session:
+            user = User(username=self.username, password=self.password)
+            session.add(user)
+            session.commit()
+            self.logged_in = True
+            return pc.redirect("/home")
+
+    def get_result(self):
+        response = openai.Completion.create(
+            model="text-davinci-002",
+            prompt=self.prompt,
+            temperature=0,
+            max_tokens=100,
+            top_p=1,
+        )
+        self.result = response["choices"][0]["text"].replace("\n", "")
+
+    def save_result(self):
+        with pc.session() as session:
+            question = Question(
+                username=self.username, prompt=self.prompt, question=self.result
+            )
+            session.add(question)
+            session.commit()
+
+
+def render_question(question):
+    return pc.tr(
+        pc.td(question.prompt),
+        pc.td(question.question),
+    )
+
+
+def home():
```

Comment:
Since this app is a bit bigger, maybe we split up each page to its own file?

Reply:
I think these demo apps can be one pagers for the most part but open to sectioning them out, I just think it may be easier for people to see these full apps in one file.

Reply:
Twitter app I will split up for Sur though